tests/func/test_add.py

```diff
@@ -59,6 +60,40 @@ def test_add(tmp_dir, dvc):
     }
 
 
+def test_add_output(tmp_dir, dvc):
+    (fd, fname) = tempfile.mkstemp()
+    os.write(fd, b"foo")
+    os.close(fd)
+
+    ret = main(["add", "--output", "foo", "bar", "baz"])
+    assert ret == 1
+
+    md5, _ = file_md5(fname)
+    stage = dvc.add(fname, output="foo")[0]
```

Comment:
You are actually implementing straight-to-cache functionality https://github.com/iterative/dvc/issues/4520 for local files here :slightly_smiling_face: We have a WIP for a general straight-to-remote, where we also use `-o` in add, they are in no way contradictory though.